lib/src/modules/gates.dart

```diff
@@ -705,6 +707,11 @@ class IndexGate extends Module with InlineSystemVerilog {
 
   /// Executes the functional behavior of this gate.
   void _execute() {
+    if (_original.width == 1) {
```

Comment:
I don't think we need this if statement, since we probably want X-prop if the `_index` is invalid, and also want to flag an error if the `_index` is out of bounds, which code below should already cover